### PR TITLE
Added CMake Action to run tilt tests on push and PR

### PR DESCRIPTION
**Pull request by xiasongh**
An example of a successful run https://github.com/xiasongh/tilt/actions/runs/4921210929/jobs/8790820396

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: CMake
+
+on:
+  push:
+    branches: [ "**" ]
```
Comment (anandj91) [inline .github/workflows/cmake.yml:5]: We only need to run the tests for pull requests. You can remove this.

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
```
Comment (anandj91) [inline .github/workflows/cmake.yml:28]: How are you installing the dependencies? For instance LLVM 11.0.0

Comment (anandj91) [reply .github/workflows/cmake.yml:28]: Can you check what is the best way to configure dependencies for the Action.

**Review by anandj91 — CHANGES_REQUESTED** [review]

File: .github/workflows/cmake.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
+    # You can convert this to a matrix build if you need cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    
+    # We use Ubuntu 20.04 so we have access to LLVM 11 dependency
```
Comment (xiasongh) [inline .github/workflows/cmake.yml:17]: Added comment on how we are getting llvm dependency

File: .github/workflows/cmake.yml (hunk 2 above)
Comment (xiasongh) [reply .github/workflows/cmake.yml:28]: We use Ubuntu 20.04, which gives us access to llvm 11. Also left a comment on how we are installing the dependency

**Review by anandj91 — APPROVED** [review]

**Pull request merged**